src/jarabe/journal/expandedentry.py

```diff
@@ -512,7 +512,19 @@ def _update_entry(self, needs_update=False):
 
         old_title = self._metadata.get('title', None)
         new_title = self._title.get_text()
-        if old_title != new_title:
+        if new_title == '' or new_title.isspace():
+            alert = Alert()
+            alert.props.title = _('Empty title')
+            alert.props.msg = _('The title cannot be empty')
+            ok_icon = Icon(icon_name='dialog-ok')
+            alert.add_button(Gtk.ResponseType.OK, _('Ok'), ok_icon)
```

Comment:
On a second thought, maybe we can have something in between prohibiting empty names, and allowing them. How about a warning instead of an alert, "The title is set to empty. Are you sure you want to proceed?", perhaps something like this :thinking: , thoughts?

Reply:
That does sound like a good idea but alerts are one major way to give additional info to the user and are usually not interactive, I don't know if they're any interactive implementations of such in sugar. @quozl does such exist?

Reply:
@chimosky I'm in favor of asking the questtion to the user, as @srevinsaju has suggested. More often than not, I'd imagine a journal entry with no name would be something they would want to correct, and was likely unintentional.

Reply:
> @chimosky I'm in favor of asking the questtion to the user, as @srevinsaju has suggested. More often than not, I'd imagine a journal entry with no name would be something they would want to correct, and was likely unintentional.

Agreed, took a look and found `sugar3.graphics.alert.ConfirmationAlert()` which is perfect for this. 

If anyone else has no objection I'll implement with `ConfirmationAlert()`.